### PR TITLE
feat(spanner): Add a DirectPath fallback integration test

### PR DESCRIPTION
**Pull request by mohanli-ml**
Add a DirectPath fallback test for spanner in Golang. The logic of the test is:

1. First wait until DirectPath traffic is observed;
2. Blackhole DirectPath net, and wait until the client fallback to use CFE;
3. Unblackhole DirectPath net, and the client should upgrade back to use DirectPath again.

The DirectPath and CFE traffic are distinguished by peer IP.

### CODE REVIEW

File: spanner/integration_test.go [hunk 1]
```diff
@@ -2957,6 +2968,52 @@ func TestIntegration_StartBackupOperation(t *testing.T) {
...
+	}
+	if len(allowDpv6Cmd) == 0 {
+		t.Fatal("-it.allowdpv6-cmd unset")
+	}
+	if len(allowDpv4Cmd) == 0 {
+		t.Fatal("-it.allowdpv4-cmd unset")
+	}
+
+	// Precondition: wait for DirectPath to connect.
+	countEnough := examineTraffic(ctx, client /*blackholeDP = */, false)
```
Comment (thiagotnunes) [inline spanner/integration_test.go:2997]: nit: `/*blackholeDP = */` should be added after the `,`

**Review by thiagotnunes — APPROVED** [review]

File: spanner/integration_test.go (hunk 1 above)
Comment (mohanli-ml) [reply spanner/integration_test.go:2997]: The problem is that gofmt will automatically move `/*blackholeDP = */` before the `,`, do you want me to remove the comment?

Comment (thiagotnunes) [reply spanner/integration_test.go:2997]: No, nevermind, maybe that is the standard.

**Comment by josephalvear** [comment]
Not interested.   
  
  
  
  
  
    
  

  
  
>

**Pull request merged**